Ajout du Bouton de suppression avec sa méthode

Arthur-Gev2023 opened the pull request:
- Add handleDeleteRequest in Dashboard Page
- Add a new const for delete in filter
- Add a new condition in requestContent.component

Arthur-Gev2023 committed:
Ajout du Bouton de suppression avec sa méthode

- Add handleDeleteRequest in Dashboard Page
- Add a new const for delete in filter
- Add a new condition in requestContent.component

robinjessondecathlon commented on src/app/component/dashboard-request.component/dasboard-request-content.component.tsx:
```diff
@@ -31,29 +33,62 @@ function RequestContent(props: { requestId: number | undefined }) {
 
   if (props.requestId === undefined || !selectedRequest) {
-    return <div>Pas de requête sélectionnée</div>
+    return (
```
ici tu dupliques du code, soit tu remets l'ancienne div `<div>Pas de requête sélectionnée</div>`, soit tu trouves un moyen de ne aps dupluqiuer

robinjessondecathlon commented on src/app/pages/dashboard/page.tsx:
```diff
@@ -2,22 +2,37 @@
+
+  const handleDeleteRequest = async (idRequest: number | undefined) => {
+    try {
+      await callApiToDeleteById(idRequest);
```
ressaye le subscribe ici

Arthur-Gev2023 committed:
Change HandleDelete Method

- change on page.tsx
- change condition on requestsContent.component

robinjessondecathlon commented on src/app/pages/dashboard/page.tsx:
```diff
@@ -4,21 +4,34 @@ import RequestContent from "@/app/component/dashboard-request.component/dasboard
-      console.log(error)
-    }
+  const handleDeleteRequest = (idRequest: number | undefined) => {
+    const deleteRequest$ = new Observable<void>((observer) => {
```
argh c'est quoi ça ?
```ts
callApiToDeleteById(idRequest).subscribe({
      next: () => {
        setDeletedRequestId(idRequest);
        setSelectedRequestId(undefined);
      },
      error: (error) => {
        console.log(error);
      },
});
```

robinjessondecathlon replied:
il faut à tout prux que tu mettes tout ça sans try catch

Arthur-Gev2023 committed:
New HandleDeleteRequest

- New HandleDeleteRequest avec subscribe sans try-catch

robinjessondecathlon commented:
tu as beaucoup de conflit, il faut que dans github desktop tu te mettes sur la branche puis tu cliques sur le bouton comme si tu allais changer de branche, en bas tu as un bouton `choose a branch to merge into ajouter-le-bouton-de-suppression`, tu cliques dessus et tu choisis main.

<img width="430" alt="image" src="https://github.com/Arthur-Gev2023/client-appl/assets/89020905/49b09bfb-3dde-4ad8-8cd5-f12e086c5975">

puis 
<img width="430" alt="image" src="https://github.com/Arthur-Gev2023/client-appl/assets/89020905/a2c4378f-0adc-4a75-b7f5-df13fba04bf3">
ça ouvre cette popup où tu peux cliquer pour ouvrir chaque conflit sur vscode
<img width="523" alt="image" src="https://github.com/Arthur-Gev2023/client-appl/assets/89020905/5f23b24c-e138-4a38-a607-5d6f96733424">

robinjessondecathlon committed:
supprime le fetchDate

robinjessondecathlon commented on src/app/pages/dashboard/page.tsx:
```diff
@@ -2,22 +2,51 @@
+
+    deleteRequest$.subscribe(
+      () => {
+        setDeletedRequestId(idRequest);
```
il faut supprimer ce useEffect vide

Arthur-Gev2023 committed:
Merge branch 'main' into ajouter-le-bouton-de-suppression-d'une-request-et-appeler-l'api

Arthur-Gev2023 committed:
Last Add Create Method Front

- Create Method on EditPage
- Change Css
- New Page : EditPage
- New Css Page for Edit Formulaire
- New Component NavBar for DashboardPage and EditPage
- New createdRequest.response for response of CreateMethod.

robinjessondecathlon committed:
pr
